poetry/core/masonry/builders/builder.py

```diff
@@ -273,6 +276,24 @@ def convert_entry_points(self):  # type: () -> dict
 
         return dict(result)
 
+    def convert_script_files(
+        self, base_path=None
+    ):  # type: (Optional[str]) -> List[Path]
+        script_files = []
+
+        for _, ep in self._poetry.local_config.get("scripts", {}).items():
+            if isinstance(ep, dict) and "source" in ep:
+                abs_path = Path.joinpath(base_path or self._path, ep["source"])
```

Comment:
```suggestion
                abs_path = (base_path or self._path).resolve()
```

If we want to get the absolute path of an object, leveraging the `pathlib.resolve()` would be more suitable. We are also using this in the [`BuildIncludeFile`](https://github.com/python-poetry/poetry-core/blob/master/poetry/core/masonry/builders/builder.py#L299).

Reply:
Awesome, I didn’t know about that!

Reply:
Ah you're right -- I gave the wrong suggestion. It should be:

```
abs_path = (base_path or self._path / ep["source"]).resolve()
```

Reply:
Updated the code, it's pretty clean now as base path is not needed